CRM/Core/Form/EntityFormTrait.php

```diff
@@ -0,0 +1,170 @@
+  | This file is a part of CiviCRM.                                    |
+  |                                                                    |
+  | CiviCRM is free software; you can copy, modify, and distribute it  |
+  | under the terms of the GNU Affero General Public License           |
+  | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+  |                                                                    |
+  | CiviCRM is distributed in the hope that it will be useful, but     |
+  | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+  | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+  | See the GNU Affero General Public License for more details.        |
+  |                                                                    |
+  | You should have received a copy of the GNU Affero General Public   |
+  | License and the CiviCRM Licensing Exception along                  |
+  | with this program; if not, contact CiviCRM LLC                     |
+  | at info[AT]civicrm[DOT]org. If you have questions about the        |
+  | GNU Affero General Public License or the licensing of CiviCRM,     |
+  | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+  +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2018
+ */
+
+trait CRM_Core_Form_EntityFormTrait {
+  /**
+   * Get entity fields for the entity to be added to the form.
+   *
+   * @var array
+   */
+  public function getEntityFields() {
+    return $this->entityFields;
+  }
+
+  /**
+   * Explicitly declare the form context.
+   */
+  public function getDefaultContext() {
+    return 'create';
+  }
+
+  /**
+   * Get entity fields for the entity to be added to the form.
+   *
+   * @var array
+   */
+  public function getDeleteMessage() {
+    return $this->deleteMessage;
+  }
+
+  /**
+   * Get the entity id being edited.
+   *
+   * @return int|null
+   */
+  public function getEntityId() {
+    return $this->_id;
+  }
+  /**
+   * If the custom data is in the submitted data (eg. added via ajax loaded form) add to form.
+   */
+  public function addCustomDataToForm() {
+    CRM_Custom_Form_CustomData::addToForm($this);
+  }
+
+  /**
+   * Build the form object.
+   */
+  public function buildQuickEntityForm() {
+    if ($this->_action & CRM_Core_Action::DELETE) {
+      $this->buildDeleteForm();
+      return;
+    }
+    $this->applyFilter('__ALL__', 'trim');
+    $this->addEntityFieldsToTemplate();
+    $this->assign('entityFields', $this->entityFields);
+    $this->assign('entityID', $this->getEntityId());
+    $this->assign('entityInClassFormat', strtolower(str_replace('_', '-', $this->getDefaultEntity())));
+    $this->assign('entityTable', CRM_Core_DAO_AllCoreTables::getTableForClass(CRM_Core_DAO_AllCoreTables::getFullName($this->getDefaultEntity())));
+    $this->addCustomDataToForm();
```

Comment:
IMO the function addCustomDataToForm should be called conditionally based on whether that entity is in the cg_extend_objects option value table.

Reply:
ok yes - will add a conditional

Reply:
@colemanw I have added a conditional - albeit in the addCustomDataToForm function